hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcServer.java

```diff
@@ -97,21 +85,16 @@ public NettyRpcServer(Server server, String name, List<BlockingServiceAndInterfa
     super(server, name, services, bindAddress, conf, scheduler, reservoirEnabled);
     this.bindAddress = bindAddress;
     this.channelAllocator = getChannelAllocator(conf);
-    EventLoopGroup eventLoopGroup;
-    Class<? extends ServerChannel> channelClass;
-    if (server instanceof HRegionServer) {
-      NettyEventLoopGroupConfig config = ((HBaseServerBase) server).getEventLoopGroupConfig();
-      eventLoopGroup = config.group();
-      channelClass = config.serverChannelClass();
-    } else {
-      int threadCount = server == null
-        ? EVENTLOOP_THREADCOUNT_DEFAULT
-        : server.getConfiguration().getInt(HBASE_NETTY_EVENTLOOP_RPCSERVER_THREADCOUNT_KEY,
-          EVENTLOOP_THREADCOUNT_DEFAULT);
-      eventLoopGroup = new NioEventLoopGroup(threadCount,
-        new DefaultThreadFactory("NettyRpcServer", true, Thread.MAX_PRIORITY));
-      channelClass = NioServerSocketChannel.class;
+    // Get the event loop group configuration from the server class if available.
+    NettyEventLoopGroupConfig config = null;
+    if (server != null) {
```

Comment:
Still better to check whether server is an instance of HBaseServerBase? Or at least add a comment to say for now, if we pass a non-null server then it must be HBaseServerBase(or HRegionServer for branch-2.x). Otherwise other developers may be confused on this direct cast.

Reply:
I will add the comment, sounds good.